src/plugins/sync_to_async.ts

```diff
@@ -0,0 +1,380 @@
+import * as types from "babel-types";
+import { compact, uniq, chain, get } from "lodash";
+
+/**
+ * Tests the name of a member expression's identifier
+ * to check whether it is a call to a UI Widgets (jQuery adapter) API
+ *
+ * @param {any} name 
+ * @returns {boolean} 
+ */
+function isUIWidgetExpression(name): boolean {
+  return /^zd((Combo)?Select)?Menu$/.test(name);
+}
+
+/**
+ * Recursively replaces expressions that refer to a given binding.
+ * This method is primarily for converting v1 api calls to wrappers
+ * to direct property access.  The replacement is predicated on an
+ * object being returned from an async/await expression for a wrapped
+ * SDK client call.
+ * 
+ * @param {any} binding 
+ */
+function replaceReferencesForBinding(binding): void {
+  let exp;
+  if (binding.referenced) {
+    const name = binding.identifier.name;
+    binding.referencePaths.forEach(p => {
+      const { exp: e, names: n } = getExpressionToReplace(p);
+      if (n.length) {
+        e.replaceWithSourceString(`${name}${"." + n.join(".")}`);
+        if (e.parentPath.isVariableDeclarator()) {
+          replaceReferencesForBinding(
+            e.parentPath.scope.bindings[e.parent.id.name]
+          );
+        } else if (
+          e.inList &&
+          (exp = e.getSibling(e.key + 1)) &&
+          exp.isFunctionExpression() &&
+          exp.node.params.length
+        ) {
+          const b = exp.scope.bindings[exp.get("params.0").node.name];
+          replaceReferencesForBinding(b);
+        }
+      }
+    });
+  }
+}
+
+/**
+ * This method extracts a complete expression that comprises a v1 api call
+ * An example would be `var email = this.ticket().requester().email()`.
+ * In the above case, the value of `exp` would be a Babel path object
+ * containing all of the nodes for `this.ticket().requester().email()`.
+ * The value of `names`, in the same case, would be `["ticket", "requester", "email"]`.
+ * 
+ * @param {any} path 
+ * @returns {{ exp: any; names: string[] }} 
+ */
+function getExpressionToReplace(path): { exp: any; names: string[] } {
+  const names = [],
+    exp = path.findParent(p => {
+      // As long as the parent expression is either a MemberExpression, or a
+      // CallExpression, keep on walking up the AST.  If we hit a parent that is
+      // in a list (`inList`), then we've likely found the complete expression.
+      // We need to explicitly exclude the "native" `bind` and `then` property names.
+      // Those two names "look" like a continuation of the same chained expression, but
+      // they're obviously not.
+      if (
+        (!p.parentPath.isMemberExpression() &&
+          !p.parentPath.isCallExpression()) ||
+        ((p.isCallExpression() && p.inList) ||
+          (p.parentPath.isMemberExpression() &&
+            /^(then|bind)$/.test(p.parent.property.name)))
```

Comment:
Not sure I understand why only `then` and `bind` can be a problem here.

Reply:
I'll add some more tests to cover this behaviour

Reply:
ea8b655